### ISSUE TITLE
Library not loaded "libreadline"?

### ISSUE THREAD

**Issue by jgillman** [hnarayanan/shpotify#68]
I'm not sure what happened, I probably updated something on my system and didn't realize it broke shpotify... I'm getting this error when I try to run it:

```
$ spotify
dyld: Library not loaded: /usr/local/opt/readline/lib/libreadline.6.dylib
  Referenced from: /usr/local/bin/bash
  Reason: image not found
```

When I look in `/usr/local/opt/readline/lib` I see libreadline, but it's `libreadline.7.dylib` (which I assume means I have version 7). I've got shpotify installed with brew.

**Comment by hnarayanan** [comment]
I have no idea, but this seems like a breakage well below the level of shpotify. Try uninstalling and reinstalling libreadline in homebrew to see if that helps.

I can't give you clearer advice off the top of my head as I am not a homebrew user and I haven't seen this particular bug before.

**Comment by jgillman** [comment]
@hnarayanan Thank you, I've tried uninstalling and reinstalling with brew to no avail but this is definitely not an issue with shpotify and perhaps inappropriate to put in an issue here

We hoping another user would have the same problem and could point me in the right direction!

**Comment by hnarayanan** [comment]
I understand. I will leave this issue open here for a couple of days. Hopefully someone else can help you in that period.

**Comment by jgillman** [comment]
[Found a solution!](https://github.com/Homebrew/homebrew-core/issues/5799#issuecomment-252953038)

**TL;DR:** `brew upgrade bash`

Hopefully this will help at least one other person searching for the same thing 😄

**Comment by lovejoy** [comment]
I use zsh` brew upgrade bash` dont fix 
` ln -s /usr/local/opt/readline/lib/libreadline.7.dylib /usr/local/opt/readline/lib/libreadline.6.dylib`
fix this

**Comment by hnarayanan** [comment]
Thanks @lovejoy. Hope your note here helps someone in the future.

**Comment by the-c0d3r** [comment]
Thank you @lovejoy, because of this bug, i've tried uninstalling and reinstalling zsh, antigen, and vim a bunch of times (on a side note, i've been using antibody ever since). I have tried symlinking the 7.dylib to 6.dylib somewhere else but it didn't work. Must be a different location. There was this one line of code from antigen.zsh that calls awk, which was the main culprit.

**Comment by ccarlile** [comment]
@lovejoy I use zsh as well, but `brew upgrade bash` fixed the issue for me anyway.

**Comment by sqhtiamo** [comment]
Thanks @lovejoy. I use zsh as well, and it really works!

**Comment by lutaoact** [comment]
@sqhtiamo It really works for me as well.

**Comment by cnees** [comment]
Upgrading bash didn't solve the problem for me. Symlinking didn't either. I ended up fixing it by copying the files from my brew installation of readline (version 7) into the folder where it said it was looking for those files.

`sudo cp /usr/local/Cellar/readline/7.0.3_1/lib/*.7.* /usr/local/opt/readline/lib/`

Use `brew info readline` to find the first path and the error message to find the second path.

**Comment by crucialfelix** [comment]
Rather than link libreadline 7 to libreadline 6 (it's the wrong version !) it's safer to find a copy of libreadline 6 on your system and symlink to that 

In my case:

    sudo ln -s /usr/local/Cellar/readline/6.3.8/lib/libreadline.6.dylib /usr/local/opt/readline/lib/libreadline.6.dylib

**Comment by zackkatz** [comment]
I ended up using `brew link readline --force` and it worked for me.

**Comment by hnarayanan** [comment]
Thank you everyone for this conversation, but I am going to lock it for this is at a level of the stack that is below shpotify.
